xbmc/windowing/X11/WinSystemX11.cpp

```diff
@@ -63,6 +63,13 @@ CWinSystemX11::~CWinSystemX11() = default;
 
 bool CWinSystemX11::InitWindowSystem()
 {
+  const char* env = getenv("DISPLAY");
+  if (!env)
```

Comment:
I prefer comparing pointers against nullptr, as that preserves the context of pointers, instead of interpreting a pointer in the context of bools

Reply:
This seems to come up every now and then. I think @yol prefers this style though. So is it just a matter of style, context, or function?

Reply:
It's your style

Reply:
As most of the time, matter of taste :)

I've quickly checked

* Google - does not specify
* WebKit - prefers this style https://webkit.org/code-style-guidelines/#zero-comparison
* HP - does not specify
* ISO C++ - prefers this style https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es87-dont-add-redundant--or--to-conditions

and most importantly

* Kodi - does not specify

 @lrusak is right though that this would be my preference. My personal opinion is that it is valid to implicitly cast a pointer to bool since the intent is quite clear and nothing surprising can happen.